components/dashboard/src/data/git-providers/unified-repositories-search-query.ts

```diff
@@ -17,15 +17,18 @@ type UnifiedRepositorySearchArgs = {
 // Combines the suggested repositories and the search repositories query into one hook
 export const useUnifiedRepositorySearch = ({ searchString, excludeProjects = false }: UnifiedRepositorySearchArgs) => {
     const suggestedQuery = useSuggestedRepositories();
-    const searchQuery = useSearchRepositories({ searchString });
+    const searchLimit = 30;
+    const searchQuery = useSearchRepositories({ searchString, limit: searchLimit });
 
     const filteredRepos = useMemo(() => {
         const flattenedRepos = [suggestedQuery.data || [], searchQuery.data || []].flat();
+
         return deduplicateAndFilterRepositories(searchString, excludeProjects, flattenedRepos);
     }, [excludeProjects, searchQuery.data, searchString, suggestedQuery.data]);
 
     return {
         data: filteredRepos,
+        hasMore: searchQuery.data?.length === searchLimit,
```

Comment:
Just to make sure: because of this, we assume that a query "has more", even if we actually just return exactly 30 responses?

Reply:
we assume it has more if we return the exact limited amount, yes.

Reply:
Should we be calculating/returning this `hasMore` from the api? Since it searches across all of your git providers it might be more accurate as we don't know how many git providers we may have searched across, which would affect how many results is the limit. Does that make sense?

Reply:
we implicitly return that by providing a limit and getting a list of a certain size returned. I tried to avoid changing the return type and too much of the API in general.
When we move this to GRPC we should combine getSuggestedRepositories and searchRepositories as well as supporting pagination.